### PR TITLE
fix: [DEL-3578]: dont get token from global context on every heartbea…

### PR DESCRIPTION
**Pull request by arpit3107**
…t call

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31724)

### CODE REVIEW

File: 350-event-server/src/main/java/io/harness/event/grpc/DelegateTokenEventServerAuthenticatorImpl.java [hunk 1]
```diff
@@ -92,7 +92,7 @@
                      .orElse(null));
 
   @Override
-  public void validateDelegateToken(String accountId, String tokenString) {
+  public void validateDelegateToken(String accountId, String tokenString, boolean isRestCall) {
```
Comment (abhinav-harness) [inline 350-event-server/src/main/java/io/harness/event/grpc/DelegateTokenEventServerAuthenticatorImpl.java:95]: Lets make it more explicit -> setTokenNameInGlobalContext

File: 270-verification/src/main/java/io/harness/app/VerificationAuthFilter.java [hunk 2]
```diff
@@ -53,6 +53,6 @@ public void filter(ContainerRequestContext containerRequestContext) {
 
   @Override
   public void validateDelegateToken(String accountId, String tokenString) {
-    delegateTokenAuthenticator.validateDelegateToken(accountId, tokenString);
+    delegateTokenAuthenticator.validateDelegateToken(accountId, tokenString, true);
```
Comment (abhinav-harness) [inline 270-verification/src/main/java/io/harness/app/VerificationAuthFilter.java:56]: we probably don't need it here

Comment (abhinav-harness) [reply 270-verification/src/main/java/io/harness/app/VerificationAuthFilter.java:56]: Ok we can have it - its harmless

**Review by Vbhatia20 — APPROVED** [review]
> Reviewed 9 of 15 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31724)* status: :shipit: complete! ✓

File: 970-grpc/src/main/java/io/harness/grpc/auth/DelegateAuthServerInterceptor.java [hunk 3]
```diff
@@ -81,7 +81,7 @@ public DelegateAuthServerInterceptor(DelegateTokenAuthenticator tokenAuthenticat
     }
     Context ctx;
     try {
-      tokenAuthenticator.validateDelegateToken(accountId, token);
+      tokenAuthenticator.validateDelegateToken(accountId, token, false);
```
Comment (gaurav-jain1) [inline 970-grpc/src/main/java/io/harness/grpc/auth/DelegateAuthServerInterceptor.java:84]: Is this the one getting called during hearbeat? and we want to disable updating global context here

**Review by gaurav-jain1 — APPROVED** [review]
Looks good..

**Pull request merged**